rusk/src/lib/chain/rusk.rs

```diff
@@ -361,20 +394,55 @@ impl Rusk {
         Ok(session)
     }
 
-    pub(crate) fn set_current_commit(&self, commit: [u8; 32]) {
+    pub(crate) fn set_current_commit(
+        &self,
+        commit: [u8; 32],
+        block_height: u64,
+    ) -> Result<()> {
         let mut tip = self.tip.write();
+
+        // If the current block height is a multiple of the epoch, we need to
+        // write the commit to the epoch commit file, and set the commit in the
+        // tip.
+        if block_height % EPOCH == 0 {
+            let epoch_commit_path = to_rusk_epoch_id_path(&self.dir);
+            write_commit_id(commit, epoch_commit_path)?;
+            tip.epoch = Some(commit);
+        }
+
         tip.current = commit;
+
+        Ok(())
     }
 
-    pub(crate) fn set_base_and_delete(&self, commit: [u8; 32]) {
+    pub(crate) fn set_base_and_delete(
+        &self,
+        commit: [u8; 32],
+        block_height: u64,
+    ) -> Result<()> {
         let mut tip = self.tip.write();
 
-        tip.current = commit;
-        tip.base = commit;
+        // If the current block height is a multiple of the epoch, we need to
+        // write the commit to the epoch commit file, and set the commit in the
+        // tip.
+        if block_height % EPOCH == 0 {
+            let epoch_commit_path = to_rusk_epoch_id_path(&self.dir);
+            write_commit_id(commit, epoch_commit_path)?;
+            tip.epoch = Some(commit);
+        }
 
-        // We will delete all commits except the new commit.
+        let current_commit = tip.current;
+        let base_commit = tip.base;
+
+        let f = commit_retain_closure(
+            [current_commit, base_commit, commit],
```

Comment:
This needs to be changed according to latest master

Reply:
Fixed